lib/livebook/file_system/s3.ex

```diff
@@ -22,20 +31,25 @@ defmodule Livebook.FileSystem.S3 do
       to have the format `*.[region].[rootdomain].com` and the region
       is inferred from that URL
 
+    * `:external_id` - the external id from Teams. By default is
+      the same of the hashed ID
+
   """
   @spec new(String.t(), String.t(), String.t(), keyword()) :: t()
   def new(bucket_url, access_key_id, secret_access_key, opts \\ []) do
-    opts = Keyword.validate!(opts, [:region])
+    opts = Keyword.validate!(opts, [:region, :external_id])
 
     bucket_url = String.trim_trailing(bucket_url, "/")
     region = opts[:region] || region_from_uri(bucket_url)
 
     hash = :crypto.hash(:sha256, bucket_url) |> Base.url_encode64(padding: false)
     id = "s3-#{hash}"
+    external_id = opts[:external_id] || id
```

Comment:
I think it's better to just keep it nil, otherwise it's not clear if it's valid external id or not.